src/main/java/org/spongepowered/common/item/inventory/lens/impl/minecraft/ContainerLens.java

```diff
@@ -0,0 +1,118 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.item.inventory.lens.impl.minecraft;
+
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryCraftResult;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+import org.spongepowered.common.item.inventory.adapter.InventoryAdapter;
+import org.spongepowered.common.item.inventory.lens.Lens;
+import org.spongepowered.common.item.inventory.lens.SlotProvider;
+import org.spongepowered.common.item.inventory.lens.comp.GridInventoryLens;
+import org.spongepowered.common.item.inventory.lens.comp.HotbarLens;
+import org.spongepowered.common.item.inventory.lens.impl.MinecraftLens;
+import org.spongepowered.common.item.inventory.lens.impl.comp.GridInventoryLensImpl;
+import org.spongepowered.common.item.inventory.lens.impl.comp.HotbarLensImpl;
+import org.spongepowered.common.item.inventory.lens.impl.comp.OrderedInventoryLensImpl;
+import org.spongepowered.common.item.inventory.lens.impl.slots.OutputSlotLensImpl;
+import org.spongepowered.common.item.inventory.lens.impl.slots.SlotLensImpl;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ContainerLens extends MinecraftLens {
+
+    // The container for reference
+    private Container container;
+    // The viewed inventories
+    private List<Lens<IInventory, ItemStack>> inventories = new ArrayList<>();
+    // In vanilla containers display the player inventory + hotbar
+    private GridInventoryLens player;
+    private HotbarLens hotbar;
+
+
+    public ContainerLens(Container container, InventoryAdapter<IInventory, ItemStack> adapter, SlotProvider<IInventory, ItemStack> slots) {
+        super(0, adapter.getInventory().getSize(), adapter, slots);
+        this.container = container;
+        this.init(slots);
+    }
+
+    @Override
+    protected void init(SlotProvider<IInventory, ItemStack> slots) {
+        System.out.print("### ContainerLens for: " + container.getClass().getName() + " ###\n");
```

Comment:
Can I prevent it from logging twice?

`new PrettyPrinter(60).add("TEST").trace(System.out);`

```
[09:52:51] [Server thread/DEBUG] [ContainerUtil]: /****************************************************************/
[09:52:51] [Server thread/DEBUG] [ContainerUtil]: /* TEST                                                         */
[09:52:51] [Server thread/DEBUG] [ContainerUtil]: /****************************************************************/
> /****************************************************************/
/* TEST                                                         */
/****************************************************************/
```

I am using `System.out.print(...)` with `\n` because is does not print out the whole `[09:52:51] [Server thread/DEBUG] [ContainerUtil]`.